lib/galaxy/webapps/galaxy/config_schema.yml

```diff
@@ -902,6 +902,18 @@ mapping:
 
           Example value 'config/disposable_email_blacklist.conf'
 
+      whitelist_file:
+        type: str
+        default: config/disposable_email_whitelist.conf
+        required: false
+        desc: |
+          E-mail domains whitelist is used for filtering out users that are using
```

Comment:
I found the wording rather confusing (I know it's based on the description of blacklist file). Also, I think the logic is incorrect (unless I'm misunderstanding..): if a user's email address domain is *not* on the list, they should be denied registration, right?
Here's a suggested wording, feel free to use any/all or adapt, etc. My only real concern is the logic.

E-mail domains whitelist is used to specify allowed email address domains. If the list is non-empty and a user attempts registration using an email address belonging to a domain that is not on the list, registration will be denied. This is a more restrictive option than <blacklist_file>, and therefore, in case <blacklist_file> is defined, <whitelist_file> will be ignored.

Also, now that I think of it, I think "more restrictive" implies that if `a` is more restrictive than `b`, then `a` takes precedence if both are defined, no? Should it be say *less restrictive* instead? Not sure here..

Reply:
Thanks for the re-wording of the documentation, it is clearer now. I will add it.

Not sure about the definition of restrictive and its priority either. This rule is more restrictive in the sense that it allows less domains (only the ones specified) than the blacklist option (which allows all but the ones specified). 

If the admin chooses to whitelist certain domains it is removing a ton of domains that are not excluded with blacklist, with the aim of "closing" the registration of new users to only the owners of an email address with the allowed domains. Then, why doing the efford of blacklisting an extensive list of domains when they are not going to be allowed any way?

I sincerelly don't know. Both priority options could be argued, so the one you want to choose is fine.

Reply:
Yep, you are right about more restrictive. However, I think the logic in the `validate_user_input.py` file may be incorrect: we don't check whitelist if blacklist is defined. So, let's say we have `whitelist` set to "foo.com" and `blacklist` set to "spam.com". Then when someone registers with "john@not-allowed-domain.com", they will be able to do that because the blacklist will not filter it out and whitelist will be ignored.

So , if both lists are defined (which makes little sense, but still..), you'd expect whitelist to take precedence and filter out everything not on the list, and blacklist to be ignored. Is this correct? If so, we just need to switch the `elif` statements.

Reply:
Yep, let's give priority to whitelist then.